modules/apps/data-engine/data-engine-taglib/test/js/data-layout-builder/components/rule-builder/editor/Editor.es.js

```diff
@@ -0,0 +1,833 @@
+
+						const actionsType = document
+							.querySelectorAll('.timeline-item')[1]
+							.querySelectorAll('.ddm-field')[2]
+							.querySelector('.option-selected');
+
+						fireEvent.click(actionsType);
+
+						fireEvent.click(getByText('value'));
+
+						await waitForElement(() => {
+							return document.querySelector('.ddm-field');
+						});
+
+						if (firstArg === 'text') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('input.ddm-field-text')
+							).toBeTruthy();
+						}
+						else if (
+							firstArg === 'select' ||
+							firstArg == 'checkbox_multiple' ||
+							firstArg == 'radio'
+						) {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.form-builder-select-field')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'date') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.date-picker')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'color') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.clay-color-picker')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'image') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.input').disabled
+							).toBeTruthy();
+						}
+						else if (
+							firstArg === 'integer' ||
+							firstArg === 'double'
+						) {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('input.form-control')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'grid') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('table')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'rich_text') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector('.cke')
+							).toBeTruthy();
+						}
+						else if (firstArg === 'document_library') {
+							expect(
+								document
+									.querySelectorAll('.timeline-item')[1]
+									.querySelectorAll('.ddm-field')[3]
+									.querySelector(
+										'.liferay-ddm-form-field-document-library'
+									)
+							).toBeTruthy();
+						}
+					}
+				);
+
+				it('show the all others fields when action type is Other field', async () => {
+					const props = defaultProps();
+					const mockIsSignedIn = jest.fn();
+
+					Liferay.ThemeDisplay.isSignedIn = mockIsSignedIn;
```

Comment:
@alinedoleron just out of curiosity, why are we mock it?

Reply:
Hi @matuzalemsteles. I mocked because the Upload/DocumentLibrary field only appears when the user is logged in. After I put this the field appeared.

Reply:
Oh okay, no problem, thanks for the information! It would be nice to add a comment above the line.